mono/mini/mini-amd64.c

```diff
@@ -1236,6 +1270,9 @@ mono_arch_set_native_call_context_ret (CallContext *ccontext, gpointer frame, Mo
 		interp_cb->frame_arg_to_data ((MonoInterpFrameHandle)frame, sig, -1, storage);
 		if (temp_size)
 			arg_set_val (ccontext, ainfo, storage);
+	} else {
+		storage = arg_get_storage (ccontext, ainfo);
+		*(gpointer *)storage = interp_cb->frame_arg_to_storage (frame, sig, -1);
```

Comment:
If I understand correctly from this conversation (https://gitter.im/mono/mono?at=5ce7bda4ad024978c61dee93) with @BrzVlad, this would duplicate work on System V ABI. We should `ifdef` Windows specific ABI

Reply:
We could ifdef, but just to clarify the behavior, when using value types as return, caller allocates memory, pass address to memory in arg1 register and callee should return the same address in the return register. So it is jus a reference to the memory that is passed over, used directly by callee and then returned back to caller. The way we handle the ainfo and call info should already abstract this away, so this is already abstracted away in ainfo and the address is already copied into the frame on call and only extracted back and returned in correct register depending on ABI.